src/parser.c

```diff
@@ -551,43 +552,70 @@ json_object *parser_sensorid_to_json(const char *key, struct l_queue *list)
 	return setdatajobj;
 }
 
-int parser_jso_setdata_to_msg(json_object *jso, knot_msg_data *msg)
+struct l_queue *parser_jso_update_to_msg_list(json_object *jso)
 {
+	json_object *json_array;
+	json_object *json_data;
 	json_object *jobjkey;
-	int sensor_id;
+	knot_msg_data *msg;
+	struct l_queue *list;
+	uint64_t i;
 	int jtype;
 	int olen;
+	uint8_t sensor_id;
 
-	/* Getting 'sensor_id' */
-	if (!json_object_object_get_ex(jso, "sensor_id", &jobjkey))
-		return -EINVAL;
+	list = l_queue_new();
+	msg = l_new(knot_msg_data, 1);
 
-	if (json_object_get_type(jobjkey) != json_type_int)
-		return -EINVAL;
+	if (!json_object_object_get_ex(jso, "data", &json_array))
+		goto fail;
```

Comment:
Not freeing `msg`

Reply:
The logic is wrong. I fixed on the force-push